build-scripts/config_common.cmake

```diff
@@ -60,17 +60,20 @@ if (CMAKE_SIZEOF_VOID_P EQUAL 8)
       set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
       set (CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_C_FLAGS} -fPIC")
     endif ()
-  else ()
-    include(CheckCCompilerFlag)
-    Check_C_Compiler_Flag(-m32 M32_OK)
-    if (M32_OK OR WAMR_BUILD_TARGET STREQUAL "X86_32")
-      add_definitions (-m32)
-      set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -m32")
-      set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -m32")
-    endif ()
+  endif()
+elseif (CMAKE_SIZEOF_VOID_P EQUAL 4)
+  include(CheckCCompilerFlag)
+  Check_C_Compiler_Flag(-m32 M32_OK)
+  if (M32_OK OR WAMR_BUILD_TARGET STREQUAL "X86_32")
+    add_definitions (-m32)
+    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -m32")
+    set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -m32")
   endif ()
```

Comment:
Should not remove the `-m32` flags in the process of branch `if (CMAKE_SIZEOF_VOID_P EQUAL 8)`: size of void pointer is 8, meaning that the host is 64-bit, and when compiling 32-bit targets in 64-bit host, e.g. compile x86-32 in ubuntu 64-bit, we need to add the `-m32` flag, or compilation error will occur.

And if `(CMAKE_SIZEOF_VOID_P EQUAL 4)`, meaning that host is 32-bit, and when compiling 32-bit targets in it, we don't need to add `-m32` flag.

Seems that we don't need to change the script, could you please check whether it works for you if there is no modification in this file?

Reply:
OK, I checked twice and probably we do not need to change this script file that mentioned above. Now, I will try again that everything is working fine with it.